src/main/java/org/generationcp/commons/service/impl/BreedingViewImportServiceImpl.java

```diff
@@ -519,106 +493,94 @@ private Integer saveAnalysisVariable(
 
-			final DmsProject study = this.studyDataManager.getProject(studyId);
-			final String programUUID = study.getProgramUUID();
-			final DataSet trialDataSet = this.getTrialDataSet(studyId);
-
-			// Used in getting the new project properties
-			final VariableTypeList variableTypeListVariates = plotDataset.getVariableTypes().getVariates();
-
-			// Create the summary statistics variable types
-			final VariableTypeList summaryStatsVariableTypeList =
-				this.createSummaryStatsVariableTypes(summaryStatsCSV, trialDataSet, variableTypeListVariates, programUUID);
-
-			final Map<Integer, String> geolocationIdToEnvironmentMap =
-				this.createGeolocationIdEnvironmentMap(summaryStatsCSV.getData().keySet(), studyId, summaryStatsCSV.getTrialHeader());
-			final List<ExperimentValues> summaryStatsExperimentValuesList =
-				this.createSummaryStatsExperimentValuesList(trialDataSet, geolocationIdToEnvironmentMap, summaryStatsCSV);
-
-			// Save project properties and experiments
-			final DmsProject project = new DmsProject();
-			project.setProjectId(trialDataSet.getId());
-			project.setStudyType(study.getStudyType());
-
-			this.studyDataManager.saveTrialDatasetSummary(project, summaryStatsVariableTypeList, summaryStatsExperimentValuesList,
-				new ArrayList<>(geolocationIdToEnvironmentMap.keySet()));
+			final Map<String, String> aliasToVariableNameMap =
+				this.daoFactory.getProjectPropertyDAO().getByProjectId(this.getPlotDataSet(studyId).getId()).stream().collect(
+					Collectors.toMap(ProjectProperty::getAlias, pp -> pp.getVariable().getName()));
+
+			final List<Integer> traitVariableIds =
+				this.daoFactory.getCvTermDao()
+					.getByNamesAndCvId(summaryStatsCSV.getTraits().stream().map(aliasToVariableNameMap::get).collect(
+						Collectors.toSet()), CvId.VARIABLES).stream()
+					.map(CVTerm::getCvTermId).collect(
+						Collectors.toList());
+
+			final AnalysisVariablesImportRequest analysisVariablesImportRequest = new AnalysisVariablesImportRequest();
+			analysisVariablesImportRequest.setAnalysisMethodNames(summaryStatsCSV.getSummaryHeaders());
+			analysisVariablesImportRequest.setVariableIds(traitVariableIds);
+			analysisVariablesImportRequest.setVariableType(VariableType.ANALYSIS_SUMMARY.getName());
+
+			// Create analysis summary variables
+			final MultiKeyMap analysisSummaryVariablesMap =
+				this.ontologyVariableService.createAnalysisVariables(analysisVariablesImportRequest,
+					MapUtils.invertMap(aliasToVariableNameMap));
+
+			final Map<String, String> environmentTrialInstanceMap =
+				this.createEnvironmentTrialInstanceMap(summaryStatsCSV.getData().keySet(), studyId, summaryStatsCSV.getTrialHeader());
+
+			final SummaryStatisticsImportRequest summaryStatisticsImportRequest =
+				this.createSummaryStatisticsImportRequest(summaryStatsCSV, environmentTrialInstanceMap, analysisSummaryVariablesMap);
+
+			// Get the means dataset if it exists.
```

Comment:
For the comments here, do you summary statistics dataset and not means dataset?

Reply:
Yup, it should be summary statistics dataset not means dataset. I'll update the comment.